### DISCUSSION TITLE
Plans for v0.14 update

### DISCUSSION THREAD

**Discussion by mikeingold** [mikeingold/LineIntegrals.jl#84] [General]
## Breaking Changes
- Rename `IntegrationAlgorithm` to `IntegrationRule`. Consolidate terminology around integration rules (versus "algorithms", "settings", etc).
  - #82
- Fully remove `derivative` and `unitdirection` functions. All `derivative` functionality was merged into `jacobian`.
  - #88 
  - #91
- Convert floating point precision option `FP` from an optional argument to a keyword argument.
  - #91

### Development/Testing Infrastructure

- Add comment blocks to each `specialized` source file explaining why each of these `Geometry`s requires specialized methods rather than using the generalized ones?
  - #89
- Add a `@testset` for `BezierCurve` with analytic solutions.
  - #92
- Implement testing with ExplicitImports.jl, resulting in some code improvements.
  - #90
- Implement testing with JuliaFormatter.jl in SciML Style, resulting in improved code style and consistency.
  - #87
- Add Downgrade workflow to check minimum version compatibility for `compat` entries.
  - #85

**Comment by mikeingold** [comment]
@JoshuaLampert let me know if there's anything I'm missing above or that you'd like to implement prior to `v0.14.0` becoming official.

**Comment by JoshuaLampert** [comment]
I'm very happy with how the internal structure evolved with this breaking release. The implementation looks pretty clean to me. We are also at 99.25 % code coverage now, which is pretty cool :partying_face: So v0.14 is good to go from my point of view. I don't have any other breaking changes in mind.

**Comment by mikeingold** [comment]
Thanks, and thanks again for all of your inputs, feedback, and brainstorming! Between the last non-breaking refactor PR that got merged into v0.13 and all of these changes, I think the codebase here has been massively improved in quality.

**Reply by mikeingold** [reply]
I just created a PR that bumps the version number out of dev status. It also adds a test that I think may get us to 100% coverage.
